packages/sso/src/account-linking.test.ts

```diff
@@ -0,0 +1,670 @@
+			await getTestInstance({
+				account: {
+					accountLinking: {
+						enabled: true,
+						trustedProviders: [], // No trusted providers - rely on domain verification
+						existingUserMode: "trusted_providers_only",
+					},
+				},
+				plugins: [
+					sso({
+						domainVerification: { enabled: true },
+					}),
+				],
+			});
+
+		const ctx = await auth.$context;
+
+		const authClient = createAuthClient({
+			plugins: [ssoClient()],
+			baseURL: "http://localhost:3000",
+			fetchOptions: { customFetchImpl },
+		});
+
+		// Create existing user
+		await authClient.signUp.email({
+			email: domainVerifiedEmail,
+			password: "password123",
+			name: "Domain User",
+		});
+
+		// Sign in to register SSO provider
+		const { headers: adminHeaders } = await signInWithTestUser();
+
+		await auth.api.registerSSOProvider({
+			body: {
+				issuer: server.issuer.url!,
+				domain: "verified-domain.com",
+				providerId: "sso-domain-verified",
+				oidcConfig: {
+					clientId: "test",
+					clientSecret: "test",
+					authorizationEndpoint: `${server.issuer.url}/authorize`,
+					tokenEndpoint: `${server.issuer.url}/token`,
+					jwksEndpoint: `${server.issuer.url}/jwks`,
+					discoveryEndpoint: `${server.issuer.url}/.well-known/openid-configuration`,
+					mapping: {
+						id: "sub",
+						email: "email",
+						emailVerified: "email_verified",
+						name: "name",
+						image: "picture",
+					},
+				},
+			},
+			headers: adminHeaders,
+		});
+
+		// Manually set domainVerified to true (simulating verification)
+		await ctx.adapter.update({
+			model: "ssoProvider",
+			where: [{ field: "providerId", value: "sso-domain-verified" }],
+			update: { domainVerified: true },
+		});
+
+		// Try SSO sign-in
+		const headers = new Headers();
+		const signInRes = await authClient.signIn.sso({
+			providerId: "sso-domain-verified",
+			callbackURL: "/dashboard",
+			fetchOptions: {
+				throw: true,
+				onSuccess: cookieSetter(headers),
+			},
+		});
+
+		// Simulate OAuth callback flow
+		let redirectLocation = "";
+		await betterFetch(signInRes.url!, {
+			method: "GET",
+			redirect: "manual",
+			onError(context) {
+				redirectLocation = context.response.headers.get("location") || "";
+			},
+		});
+
+		if (redirectLocation) {
+			await betterFetch(redirectLocation, {
+				method: "GET",
+				headers,
+				customFetchImpl,
+				onError(context) {
+					redirectLocation = context.response.headers.get("location") || "";
+				},
+			});
+		}
+
+		// Should NOT contain error - domain is verified
+		expect(redirectLocation).not.toContain("error");
+		expect(redirectLocation).toContain("dashboard");
+
+		// Verify SSO account was linked
+		const accounts = await ctx.adapter.findMany({
+			model: "account",
+			where: [{ field: "providerId", value: "sso-domain-verified" }],
+		});
+		expect(accounts.length).toBe(1);
+	});
+});
+
+describe("SSO Account Linking - configuration tests", () => {
```

Comment:
These tests feel like they don't really belong to SSO, but rather to the account linking feature as a whole?